homeassistant/components/matter/light.py

```diff
@@ -144,6 +150,22 @@ async def _set_color_temp(
         color_temp_mired = color_util.color_temperature_kelvin_to_mired(
             color_temp_kelvin
         )
+        # clamp to supported range
+        max_mireds = (
+            self.get_matter_attribute_value(
+                clusters.ColorControl.Attributes.ColorTempPhysicalMaxMireds
+            )
+            or DEFAULT_MIREDS_MAX
+        )
+        min_mireds = (
+            self.get_matter_attribute_value(
+                clusters.ColorControl.Attributes.ColorTempPhysicalMinMireds
+            )
+            or DEFAULT_MIREDS_MIN
+        )
+        color_temp_mired = min(color_temp_mired, max_mireds)
+        color_temp_mired = max(color_temp_mired, min_mireds)
+
         await self.send_device_command(
             clusters.ColorControl.Commands.MoveToColorTemperature(
                 colorTemperatureMireds=color_temp_mired,
```

Comment:
Just to make sure I understand everything:
In `_update_from_device` below, we now correctly set `_attr_max_color_temp_kelvin` and `_attr_min_color_temp_kelvin`, which is used ("by the frontend") to determine what values are allowed with the new constant `DEFAULT_MIREDS_MIN`/`DEFAULT_MIREDS_MAX` limit in case the device reports something bad.

But here, when setting the color temperature on the device, we still clamp the values to the range where they should already be, no? Like, we set `_attr_max_color_temp_kelvin` and `_attr_min_color_temp_kelvin`, so we should never get values outside that range?

Reply:
Ok, confirmed that Home Assistant's base light code does not clamp the color temp values based on the min/max values we set.
So, this is only an issue when providing invalid values via service calls, since the frontend should (with the other changes in this PR) only display the correct range, regardless of what the light provides.

At least with ZHA, this is also an issue. I feel like this should be handled by the base light integration instead? But it's also kind of on the user if they're providing invalid values via service calls 😄 

For now, I think this should be fine..